### PR TITLE
VS2017 improvements

### PR DESCRIPTION
**Pull request by d122y**
This fixes compilation errors in VS2017

### CODE REVIEW

File: LiteCore/Database/SequenceTracker.cc [hunk 1]
```diff
@@ -217,7 +217,8 @@ namespace litecore {
                     external = i->external;
                 else if (i->external != external)
                     break;
-                changes[n++] = {i->docID, i->revID, i->sequence, i->bodySize};
```
Comment (snej) [inline LiteCore/Database/SequenceTracker.cc]: I'd rather avoid the extra copying. Does `changes[n++] = Change{i->docID, i->revID, i->sequence, i->bodySize}` work?

File: LiteCore/Support/PlatformIO.hh [hunk 2]
```diff
@@ -22,6 +22,7 @@
     #define strcasecmp  _stricmp
     #define fdclose ::_close
 
+struct stat;
```
Comment (snej) [inline LiteCore/Support/PlatformIO.hh]: Please indent 4 spaces and put a blank line after.

File: LiteCore/Database/SequenceTracker.cc (hunk 1 above)
Comment (borrrden) [reply LiteCore/Database/SequenceTracker.cc]: It does not.  This seems like a compiler error, unless something in C++14 prohibits this.  The issue comes from the mismatch of types between i->docID (alloc_slice) and docID on change (slice).  A copy constructor exists between alloc_slice and slice but it seems it is not getting hit with the initializer list.  For example, this works:

```c++
slice docID = i->docID;
slice revID = i->revID;
changes[n++] = { docID, revID, i->sequence, i->bodySize };
```

File: LiteCore/Support/PlatformIO.hh [hunk 3]
```diff
@@ -22,6 +22,8 @@
     #define strcasecmp  _stricmp
     #define fdclose ::_close
 
+    struct stat;
```
Comment (d122y) [inline LiteCore/Support/PlatformIO.hh]: @snej , updated

**Comment by borrrden** [comment]
I just realized that this PR is on the db021 branch.  It should be on dev instead.  Is there a way to modify this PR to do so?

**Comment by d122y** [comment]
@borrrden, created another PR #374 to dev. Feel free to close this one if it's easier

**Pull request closed without merging**